SYCL/ESIMD/api/functional/mutator.hpp

```diff
@@ -0,0 +1,87 @@
+//===-- mutator.hpp - This file provides common function and classes to mutate
+//      reference data. ---------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides common function and classes to mutate reference data.
+///
+//===----------------------------------------------------------------------===//
+
+#include "value.hpp"
+#include <sycl/sycl.hpp>
+
+#include <algorithm>
+#include <vector>
+
+#pragma once
+
+namespace esimd_test::api::functional {
+
+// Provides base functions to mutate value.
+namespace mutator {
+
+template <typename T> class Add {
```

Comment:
```suggestion
// Replace specific reference values to the bigger ones, so we can safely
// substract `value_for_mutation` later
template <typename T> class Add {
```
That's what we discussed earlier, but the logic below doesn't fit such description, because:

- instead of disabling UB for "substraction" operation by changing the set of input values from `[min..max]` to `[min+val..max]`, by only handling the `min` border
- we are just adding `val` for every reference value, effectively disabling some corner case checks (`denorm_min` for example)

Please verify the logic for mutators. Feel free to rename them if the naming suggested feels confusing - for example `mutator::for_substraction` instead of `mutator::add`

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/789/commits/7cc2f674c94df0416edeeb133af0de198829cfb6#diff-fa18495d278b576095ad26f3fbce0799c62e2302a8837d4e0a289c922d0dce74L34-L40)

Also mutators has been renamed